enhancements/ingress/route-secret-injection-for-external-certificate-management.md

```diff
@@ -0,0 +1,375 @@
+in abondoning the cert/key pair that was generated via cert-manager and resort to
+using the default certificates that were initially generated by the cluster-ingress-operator
+i.e. clearing the `.spec.tls` on Route. 
+
+**Deleting the generated secret that contains the serving cert/key pair**
+When the end user deletes the secret containing the certificate data, the `.spec.tls` on 
+Route will be cleared. This will result in the route serving default certificates. In such
+a scenario the annotation that was added by the user will be automatically deleted by the 
+secret-injector controller. This is only in the case where the user is directly creating 
+routes instead of relying on the Ingresses.
+
+#### Variation [optional]
+
+N.A
+
+### API Extensions
+
+A new annotation of type `string` to be introduced as part of the integration
+to support third-party certificate management systems,
+
+```yaml
+annotations:
+  route.openshift.io/tls-secret-reference: <secret-name>
+```
+
+*Note*: The default value would be N/A.
+
+This annotation can be applied by the end user on Route CR. The controller that 
+will be introduced as part of this enhancement will be responsible in
+process this annotation on the Route CR.
+
+The Route API will also be updated to denote certificate status under `.status`,
+
+```go
+
+// These are valid conditions of pod.
+const (
+
+    // ....
+    
+    // RouteCertificateDefaultType denotes that the user has not provided 
+    // any custom certificate and the default generated certificate is being
+    // served on the route.
+    RouteCertificateDefaultType RouteIngressConditionType = "Default"
+
+    // RouteCertificateCustomType denotea that there is a custom certificate 
+    // being served on the route. 
+    RouteCertificateCustomType RouteIngressConditionType = "Custom"
+
+    // RouteCertificateManagedType denotes that the annotation is in use to 
+    // inject a third-party managed secret containing the certificate that 
+    // is being served on the route.
+    RouteCertificateManagedType RouteIngressConditionType = "Managed"
+
+)
+
+```
+
+#### Variation
+
+##### Alternative to the tls-secret-reference annotation
+
+***Note***: This idea has been dropped in favor of a secret reference through an annotation.
+
+The alternative to the annotation was to provide a new API field in the Route API,
+through which the user could provide the certificate reference.
+
+```go
+type CertificateReference struct {
+	// certificate resource name.
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+}
+
+Type RouteSpec struct {
+	// ...
+
+	CertificateRef CertificateReference `json:"certificateRef,omitempty" protobuf:"bytes,5,opt,name=certificateRef"`
+}
+```
+
+The reasoning for the field was mainly for future proofing the API to handle integrations
+with other third-party certificate management systems. But this API introduces inconsistency
```

Comment:
Can you please elaborate on which inconsistency turned us way from this alternative?

Reply:
Addressed